nextcord/user.py

```diff
@@ -143,8 +153,15 @@ def default_avatar(self) -> Asset:
         """:class:`Asset`: Returns the default avatar for a given user.
 
         This is calculated by the user's discriminator.
+
+        ..versionchanged:: 2.6
+            Added handling for the new username system for users without a discriminator.
         """
-        return Asset._from_default_avatar(self._state, int(self.discriminator) % len(DefaultAvatar))
+        if self.discriminator != "0":
+            avatar_index = (self.id >> 22) % len(DefaultAvatar)
+        else:
+            avatar_index = int(self.discriminator) % 5
```

Comment:
Possible logic regression, this used to use the length of default avatar?

Reply:
It did, yes, The reason it's hardcoded is because for people who have not migrated to the new usernames, it should still be `5`, and getting the length of DefaultAvatar is going to return `6`
![image](https://github.com/nextcord/nextcord/assets/80943664/ff8559ae-4a85-499a-b81a-d3aca2d552cf)